Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Clickhouse/Ride.hs

```diff
@@ -123,3 +129,85 @@ getEarningsByIds rideIds = do
     [] -> pure 0
     [earnings] -> pure $ fromMaybe 0 earnings
     _ -> throwError $ InternalError "getEarningsByDriver query returns more than 1 response"
+
+getCompletedRidesStatsByIds ::
+  CH.HasClickhouseEnv CH.APP_SERVICE_CLICKHOUSE m =>
+  [Id DRide.Ride] ->
+  m (Int, Int, Int)
+getCompletedRidesStatsByIds rideIds = do
+  res <-
+    CH.findAll $
+      CH.select_
+        ( \ride -> do
+            let earnings = CH.sum_ ride.fare
+            let distanceTravelled = CH.sum_ ride.chargeableDistance
+            let count = CH.count_ ride.id
+            CH.groupBy ride.status $ \_ -> do
+              (earnings, distanceTravelled, count)
+        )
+        $ CH.filter_
+          ( \ride _ ->
+              ride.status CH.==. Just DRide.COMPLETED
+                CH.&&. ride.id `in_` rideIds
+          )
+          (CH.all_ @CH.APP_SERVICE_CLICKHOUSE rideTTable)
+  case res of
+    [] -> pure (0, 0, 0)
+    [(earnings, distanceTravelled, count)] -> pure (fromMaybe 0 earnings, fromMaybe 0 distanceTravelled, count)
+    _ -> throwError $ InternalError "getEarningsByDriver query returns more than 1 response"
+
+getCompletedRidesStatsByIdsAndDriverId ::
+  CH.HasClickhouseEnv CH.APP_SERVICE_CLICKHOUSE m =>
+  [Id DRide.Ride] ->
+  Id DP.Person ->
+  m (Int, Int, Int, NominalDiffTime)
+getCompletedRidesStatsByIdsAndDriverId rideIds driverId = do
+  res <-
+    CH.findAll $
+      CH.select_
+        ( \ride -> do
+            let earnings = CH.sum_ ride.fare
+            let distanceTravelled = CH.sum_ ride.chargeableDistance
+            let count = CH.count_ ride.id
+            let duration = CH.sum_ (CH.diffUTCTime ride.updatedAt ride.createdAt)
+            CH.groupBy ride.status $ \_ -> do
+              (earnings, distanceTravelled, count, duration)
```

Comment:
if you use only one status `DRide.COMPLETED`, then grouping is not required, you can use aggregate function:

`CH.aggregate (earnings, distanceTravelled, count, duration)`